mlflow/pyspark/ml/__init__.py

```diff
@@ -218,18 +346,79 @@ def _log_pretraining_metadata(estimator, params):
         if params and isinstance(params, dict):
             estimator = estimator.copy(params)
 
+        param_map = _get_instance_param_map(estimator)
+        if isinstance(estimator, Pipeline):
+            pipeline_hierarchy = _get_pipeline_stage_hierarchy(estimator)
+            try_mlflow_log(
+                mlflow.log_dict, pipeline_hierarchy, artifact_file="pipeline_hierarchy.json"
+            )
+
+        if _is_parameter_search_estimator(estimator):
+            tuning_param_maps = _get_tuning_param_maps(estimator)
+
+            try_mlflow_log(mlflow.log_dict, tuning_param_maps, 'estimator_param_maps.json')
+
+            if isinstance(estimator.getEstimator(), Pipeline):
+                estimator_pipeline_hierarchy = _get_pipeline_stage_hierarchy(
+                    estimator.getEstimator())
+                try_mlflow_log(
+                    mlflow.log_dict,
+                    estimator_pipeline_hierarchy,
+                    artifact_file="estimator_pipeline_hierarchy.json"
+                )
+
         # Chunk model parameters to avoid hitting the log_batch API limit
         for chunk in _chunk_dict(
-            _get_instance_param_map(estimator), chunk_size=MAX_PARAMS_TAGS_PER_BATCH,
+            param_map, chunk_size=MAX_PARAMS_TAGS_PER_BATCH,
         ):
             truncated = _truncate_dict(chunk, MAX_ENTITY_KEY_LENGTH, MAX_PARAM_VAL_LENGTH)
             try_mlflow_log(mlflow.log_params, truncated)
 
         try_mlflow_log(mlflow.set_tags, _get_estimator_info_tags(estimator))
 
     def _log_posttraining_metadata(estimator, spark_model, params):
-        # TODO: Log nested runs for spark ml tuning estimators
-        #   (CrossValidator/TrainValidationSplit)
+
+        if _is_parameter_search_estimator(estimator):
+            try:
+                # Fetch environment-specific tags (e.g., user and source) to ensure that lineage
+                # information is consistent with the parent run
+                child_tags = context_registry.resolve_tags()
+                child_tags.update({MLFLOW_AUTOLOGGING: AUTOLOGGING_INTEGRATION_NAME})
+                _create_child_runs_for_parameter_search(
+                    parent_estimator=estimator,
+                    parent_model=spark_model,
+                    parent_run=mlflow.active_run(),
+                    child_tags=child_tags
+                )
+            except Exception as e:
+                import traceback
+                msg = (
+                    "Encountered exception during creation of child runs for parameter search."
+                    " Child runs may be missing. Exception: {}".format(traceback.format_exc())
+                )
+                _logger.warning(msg)
+
+            estimator_param_maps = _get_tuning_param_maps(estimator)
+
```

Comment:
@WeichenXu123 Can we also log the best parameters to the parent run after the hyperparameter sweep is complete? We do this for `sklearn` and MLflow + hyperopt, and I think it would be a nice addition here.

As an example, see:

```
import mlflow
mlflow.sklearn.autolog()

from sklearn import svm, datasets
from sklearn.model_selection import GridSearchCV

iris = datasets.load_iris()
parameters = {"kernel": ("linear", "rbf"), "C": [1, 5, 10]}
svc = svm.SVC()
clf = GridSearchCV(svc, parameters, n_jobs=1, return_train_score=True, refit=True)
clf.fit(iris.data, iris.target)
```

which logs the following optimal params to the parent:

```
{'best_C': '5', 'best_kernel': 'rbf'}
```